x-pack/beatless/core/coordinator.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package core
+
+import (
+	"context"
+
+	"github.com/joeshaw/multierror"
+
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// Runner is the interface that the coordinator will follow to manage a function goroutine.
+type Runner interface {
+	Run(context.Context) error
+}
+
+// Coordinator takes care of managing the function goroutine, it receives the list of functions that
+// need to be executed and manage the goroutine.  If an error happen and its not handled by the
+// function, we assume its a fatal error and we will
+// stop all the other goroutine and beatless will terminate.
+type Coordinator struct {
+	log     *logp.Logger
+	runners []Runner
+}
+
+// NewCoordinator create a new coordinator objects receiving the clientFactory and the runner.
+func NewCoordinator(log *logp.Logger,
+	runners ...Runner,
+) *Coordinator {
+	if log == nil {
+		log = logp.NewLogger("coordinator")
+	}
+	return &Coordinator{log: log, runners: runners}
+}
+
+// Start starts each functions into an independant goroutine and wait until all the goroutine are
+// stopped to exit.
+func (r *Coordinator) Start(ctx context.Context) error {
```

Comment:
I would rather name this function `Run`. `Start` implies there is a `Stop` function to call.